src/ml_gym/gym/jobs.py

```diff
@@ -120,33 +133,52 @@ def _evaluation_step(self, device: torch.device) -> List[EvaluationBatchResult]:
         return evaluation_results
 
     def run_checkpointing(self, checkpoint_instruction: CheckpointingInstruction):
-        # TODO use self.gs_api_client to make the calls. Note that some of the endpoints are also still missing for that...
-        
-        pass
-        # if checkpoint_instruction.save_current:
-        #     self._experiment_status_logger.log_checkpoint(epoch=self.current_epoch,
-        #                                                   model_state_dict=self.model.state_dict(),
-        #                                                   optimizer_state_dict=self.optimizer.state_dict(),
-        #                                                   lr_scheduler_state_dict=self.lr_scheduler.state_dict(),
-        #                                                   stateful_components_state_dict=self.get_state())
-        # for epoch in checkpoint_instruction.checkpoints_to_delete:
-        #     print(f"epoch to delete: {epoch}")
-        #     self._experiment_status_logger.log_checkpoint(epoch=epoch,
-        #                                                   model_state_dict=None,
-        #                                                   optimizer_state_dict=None,
-        #                                                   lr_scheduler_state_dict=None,
-        #                                                   stateful_components_state_dict=None)
+        """ 
+        Create and Delete checkpoints for each epoch in experiments.
+
+        :params:
+            checkpoint_instruction: CheckpointingInstruction object
+        """
+
+        if checkpoint_instruction.save_current:
+            payload_dict = {"epoch": self.current_epoch,
+                            "model": self.model.state_dict(),
+                            "optimizer": self.optimizer.state_dict(),
+                            "lr_scheduler": self.lr_scheduler.state_dict(),
+                            "stateful_components": self.get_state()}
+
+            self.gs_api_client.post_checkpoint_resource_call(
```

Comment:
the function names were changed because there were same function names with different parameters and return type making it difficult to follow sometimes.
Reverted to name given.